SharedSources/MediaLibraryModel/MediaModel.swift

```diff
@@ -73,3 +73,75 @@ extension VLCMLMedia {
         return integer == 0
     }
 }
+
+// MARK: - CoreSpotlight
+
+extension VLCMLMedia {
+    func coreSpotlightAttributeSet() -> CSSearchableItemAttributeSet {
+        let attributeSet = CSSearchableItemAttributeSet(itemContentType: "public.audiovisual-content")
+        attributeSet.title = title
+        attributeSet.metadataModificationDate = Date()
+        attributeSet.addedDate = Date()
+        attributeSet.duration = NSNumber(value: duration() / 1000)
+        attributeSet.streamable = 0
+        attributeSet.deliveryType = 0
+        attributeSet.local = 1
+        attributeSet.playCount = NSNumber(value: playCount())
+        if isThumbnailGenerated() {
+            attributeSet.thumbnailData = UIImage(contentsOfFile: thumbnail.path)?.jpegData(compressionQuality: 0.9)
+        }
+        attributeSet.codecs = codecs()
+        attributeSet.languages = langugaes()
+        for track in audioTracks {
+            attributeSet.audioBitRate = NSNumber(value: track.bitrate())
+            attributeSet.audioChannelCount = NSNumber(value: track.nbChannels())
+            attributeSet.audioSampleRate = NSNumber(value: track.sampleRate())
+        }
+        if let albumTrack = albumTrack {
+           // attributeSet.genre = albumTrack.genre.name
+            attributeSet.artist = albumTrack.artist.name
+            attributeSet.audioTrackNumber = NSNumber(value:albumTrack.trackNumber())
+            attributeSet.album = albumTrack.album.title
+        }
+
+        return attributeSet
+    }
+
+    func codecs() -> [String] {
+        var codecs = [String]()
+
+        for track in videoTracks where track.codec != nil {
+             codecs.append(track.codec)
+        }
+        for track in audioTracks where track.codec != nil {
+            codecs.append(track.codec)
+        }
+//        for track in subtitleTracks where track.codec != nil {
```

Comment:
Why is this commented out?

Reply:
because the subtitle tracks are going to be exposed with the new medialibrary update